tools/verifygitlog.py

```diff
@@ -100,11 +113,40 @@ def run(args):
         sys.exit(1)
 
 
+def run_check_message(filename):
+    verbose("checking commit message from", filename)
+    errors = []
+    warnings = []
+
+    def error(err):
+        errors.append(err)
+
+    def warning(err):
+        warnings.append(err)
+
+    with open(args[-1]) as f:
+        lines = [l.rstrip("\r\n") for l in f]
+        verify_message_body(lines, error, warning)
+
+    for err in errors:
+        print("error:", err)
+    for err in warnings:
+        print("warning:", err)
+
+    return any(errors), any(warnings)
+
+
 def show_help():
-    print("usage: verifygitlog.py [-v -n -h] ...")
+    print("usage: verifygitlog.py [-v -n -h --check-message] ...")
     print("-v  : increase verbosity, can be speficied multiple times")
     print("-n  : do not print multi-line suggestions")
     print("-h  : print this help message and exit")
+    print(
+        "--check-message : Pass a single argument which is a file containing a candidate commit message"
```

Comment:
I think this should be `--check-file` to indicate it reads from a file rather than git sha's.  (And because this tool always checks messages, regardless of where they come from.)

Also, feel free to switch to use `argparse`, but only if it's easier, don't want to spend too much time on this simple script!

Reply:
@dpgeorge Thanks for the review. Done, and rebased!

I had a quick look at argparse, but it's a bit tricky - not least because it supports passing arbitrary command line options through to `git log`. I'm sure it's possible, but I'm going to opt to kick that can down the road until next time a feature needs to be added here. :)